src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -674,10 +698,11 @@ public static HttpServer provideHttpServer(
             ScheduledExecutorService executor,
             @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_HOST) String host,
             @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_PORT) int port,
-            @Named(HTTP_SERVER_SSL_CTX) Optional<SSLContext> sslContext) {
+            @Named(HTTP_SERVER_SSL_CTX) Optional<SSLContext> sslContext,
+            @Named(ConfigModule.CRYOSTAT_AGENT_TLS_ENABLED) boolean tlsEnabled) {
         try {
             HttpServer http;
-            if (sslContext.isEmpty()) {
+            if (sslContext.isEmpty() || !tlsEnabled) {
```

Comment:
This is for configuring the Agent's own built-in webserver, whereas the issue and the rest of the PR is about the Agent's HTTP client connection to the Cryostat server. I think this part should be removed - these are two different contexts and should be controlled separately.